### ISSUE TITLE
AABB Collision Resolution

### ISSUE THREAD

**Issue by HeadMerchant** [MIT-Video-Game-Orchestra/vgo_game#17]
The current Axis-Aligned Bounding Box implementation doesn't properly resolve collisions. 
A more robust implementation should offer:
- normal vector of the collision edge (used for negating/reflecting velocity perpendicular to the surface)
- properly position colliding objects (objects that can move should be just barely touching)

In addition, if we plan on having fast-moving projectiles later on, I recommend looking into continuous collision detection via swept AABBs, but let's focus on getting regular collision resolutions working first.

Recommended reading:
https://noonat.github.io/intersect/

**Comment by HeadMerchant** [comment]
Also, I can't self-assign this issue, but I'll take it.

**Comment by larryoatmeal** [comment]
Yes, this would be great, it is similar to #11, might want to see if @CometsTech has some thoughts on it as well
